packages/fast-components-react-base/src/auto-suggest/auto-suggest.tsx

```diff
@@ -0,0 +1,523 @@
+                focusedItem: null,
+            });
+        } else {
+            this.setState({
+                focusedItem: newSelection[0],
+            });
+            this.updateValue(newSelection[0].value);
+        }
+    };
+
+    /**
+     * Handles clicks
+     */
+    private handleInputRegionClick = (e: React.MouseEvent): void => {
+        if (this.props.disabled || e.defaultPrevented) {
+            return;
+        }
+        this.toggleMenu(true);
+    };
+
+    /**
+     * Handles item invoked
+     */
+    private handleItemInvoked = (item: ListboxItemProps): void => {
+        this.invoke(item.value, item);
+    };
+
+    /**
+     * Invokes the auto-select component
+     */
+    private invoke = (value: string, item: ListboxItemProps): void => {
+        if (typeof this.props.onInvoked === "function") {
+            this.props.onInvoked(value, item);
+        }
+    };
+
+    /**
+     * Handles value changes from input element
+     */
+    private handleChange = (e: React.ChangeEvent): void => {
+        const newValue: string = (e.target as HTMLInputElement).value;
+        this.updateValue(newValue);
+    };
+
+    /**
+     * Update the currentValue of the component
+     */
+    private updateValue = (newValue: string): void => {
+        if (typeof this.props.onValueChange === "function") {
+            this.props.onValueChange(newValue);
+        }
+
+        if (this.props.value !== undefined || newValue !== this.state.value) {
+            this.toggleMenu(true);
+            this.setState({
+                value: newValue,
+            });
+        }
+    };
+
+    /**
+     * Handles input region key events
+     */
+    private handleInputRegionKeydown = (e: React.KeyboardEvent<Element>): void => {
+        if (this.props.disabled || e.defaultPrevented) {
+            return;
+        }
+
+        switch (e.keyCode) {
+            case KeyCodes.enter:
+                this.invoke(this.state.value, null);
+                break;
+
+            case KeyCodes.escape:
+                this.toggleMenu(false);
+                break;
+
+            case KeyCodes.arrowDown:
+                this.focusOnMenu(1);
+                break;
+
+            case KeyCodes.arrowUp:
+                this.focusOnMenu(-1);
+                break;
+
+            default:
+                if (e.target instanceof HTMLInputElement) {
+                    const newValue: string = (e.target as HTMLInputElement).value;
+                    this.updateValue(newValue);
+                    this.focusOnInput();
+                    break;
+                }
+        }
+    };
+
+    /**
+     * Handles menu key events
+     */
+    private handleMenuKeydown = (e: React.KeyboardEvent<Element>): void => {
+        if (this.props.disabled || e.defaultPrevented) {
+            return;
+        }
+        switch (e.keyCode) {
+            case KeyCodes.escape:
+                this.toggleMenu(false);
+                break;
+
```

Comment:
what type of element is focused here? If it is a div, you will likely want to call `e.preventDefault()`, otherwise when this component is on pages that overflow the window vertically the down/up arrow will also scroll the page.

Reply:
ok

Reply:
That revealed some issues with my end of list code, fixed.  This could be an issue in other places as I'm trying to only preventDefault where I absolutely have to.